src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -0,0 +1,293 @@
+{
+    PerformanceConfigHipImplicitGemmFwdXdlops pp;
+    pp.HeuristicInit(ctx);
+    return pp;
+}
+
+bool ConvHipImplicitGemmFwdXdlops::IsValidPerformanceConfig(
+    const ConvolutionContext& ctx, const PerformanceConfigHipImplicitGemmFwdXdlops& config) const
+{
+    return config.IsValid(ctx);
+}
+
+PerformanceConfigHipImplicitGemmFwdXdlops
+ConvHipImplicitGemmFwdXdlops::Search(const ConvolutionContext& ctx,
+                                     const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, invoke_ctx);
+}
+
+size_t ConvHipImplicitGemmFwdXdlops::GetWorkspaceSize(const ConvolutionContext& ctx) const
+{
+    std::ignore = ctx;
+    return 0;
+}
+
+bool ConvHipImplicitGemmFwdXdlops::IsApplicable(const ConvolutionContext& ctx) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    return false;
+#else
+    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_IMPLICIT_GEMM_HIP_FWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
+    if(!ctx.use_hip_kernels)
+        return false;
+    if(!(ctx.conv_problem.GetInDataType() == miopenInt8 &&
+         ctx.conv_problem.GetWeightsDataType() == miopenInt8 &&
+         ctx.conv_problem.GetOutDataType() == miopenInt8))
+        return false;
+    if(!ctx.direction.IsForward())
+        return false;
+    if(!ctx.Is2d())
+        return false;
+    if(ctx.GetStream().GetDeviceName() != "gfx908")
+        return false;
+    if(!ctx.IsLayoutNHWC())
+        return false;
+
+    std::vector<DeviceConvFwdPtr_t> conv_ptrs;
+    add_device_conv2d_fwd_xdl_nhwc_kyxc_nhwk_int8_instances_t(conv_ptrs);
+    assert(!conv_ptrs.empty());
+    const auto args = CKArgs{ctx};
+    if(!std::all_of(args.strides.begin(), args.strides.end(), [&](auto x) { return x == 1; }))
+        return false;
+
+    for(auto& conv_ptr : conv_ptrs)
+    {
+        auto argument_ptr = conv_ptr.MakeArgumentPointer(nullptr,
+                                                         nullptr,
+                                                         nullptr,
+                                                         args.N,
+                                                         args.K,
+                                                         args.C,
+                                                         args.input,
+                                                         args.filter,
+                                                         args.input,
+                                                         args.strides,
+                                                         args.dilation,
+                                                         args.lPadding,
+                                                         args.rPadding);
+        if(conv_ptr.IsSupportedArgument(argument_ptr.get()))
+            return true;
+    }
+    return false;
```

Comment:
[Suspicion] A mistake in logic?
```suggestion
        if(!conv_ptr.IsSupportedArgument(argument_ptr.get()))
            return false;
    }
    return true;
```

Reply:
The purpose is to break out of the loop sooner. If we can find at least a single kernel in the vector that is applicable, then the solver is applicable and we are done! .

Reply:
Resolved!